README.md

```diff
@@ -332,6 +333,60 @@ to exist for the lifetime of the view controller object:
 
 ```
 
+### Error Handling ###
+
+The emergence of `try / catch` in Swift 2 has added powerful ways to define and return errors when something fails. The emergence of `ErrorType`
+as well for defining errors makes error definitions much more convenient over the cumbersome `NSError`. Because of this, for functions where operations can fail, you should always
+define it as `throws` and return a well-defined `ErrorType`.
+
+Consider the following contrived example:
+
+```swift
+
+func multiplyEvens(evenNumber: Int) -> Int? {
+	guard evenNumber % 2 == 0 else {
+		return nil
+	}
+	
+	return evenNumber * 2
+}
+
+```
+
+The function above fails because it only expects evens and returns an optional if that is violated. While this works and is simple, it
+is more Objective-C than Swift in its composition. For Swift, instead consider refactoring it as follows:
+
+```swift
+
+internal enum NumberError: ErrorType {
+	case NotEven
+}
+
+func multiplyEvens(evenNumber: Int) throws -> Int {
+	guard evenNumber % 2 == 0 else {
+		throw NumberError.NotEven
+	}
+	
+	return evenNumber * 2
+}
+
+```
+
+The above, while slightly more cumbersome, is much more explicit and helpful to the caller than the previous. Further, try/catch semantics
```

Comment:
I would add an example where you use the power of throwing multiple error cases to demonstrate the ability to have clean error handling when multiple error types can occur